### ISSUE TITLE
Transpose/Adjoint vectors need a broadcast style

### ISSUE THREAD

**Issue by mbauman** [JuliaLang/julia#32289]
LinearAlgebra implements broadcasting for transposed/adjoint rows by specializing the `broadcast` method itself for a handful of combinations of types instead of working within the Broadcast system directly.  This leads to the following behaviors:

```julia
julia> (1:3)' * 3
1×3 LinearAlgebra.Adjoint{Int64,Array{Int64,1}}:
 3  6  9

julia> (1:3)' .* 3
1×3 Array{Int64,2}:
 3  6  9

julia> broadcast(*, (1:3)', 3)
1×3 LinearAlgebra.Adjoint{Int64,Array{Int64,1}}:
 3  6  9

julia> broadcast(*, (1:3)', Ref(3))
1×3 Array{Int64,2}:
 3  6  9

julia> broadcast(*, (1:3)', (1:3)')
1×3 LinearAlgebra.Adjoint{Int64,Array{Int64,1}}:
 1  4  9

julia> broadcast(*, (1:3)', transpose(1:3))
1×3 Array{Int64,2}:
 1  4  9

julia> broadcast(*, transpose(1:3), transpose(1:3))
1×3 LinearAlgebra.Transpose{Int64,Array{Int64,1}}:
 1  4  9
```

Implementing the style will allow us to more consistently do what we want here — preserve the transposed row — but of course it'll be a minor change in the returned array types.  We'll need to decide if, e.g., `identity.(sparse(1:3)')` should be a sparse matrix or an adjoint sparse vector.

(This is marked with a TODO in the code but I cannot find a corresponding issue.)

**Comment by vchuravy** [comment]
One of the challenging aspects is that the broadcast style of parent should be honoured as well. Since Transpose can be used to wrap arbitrary subtypes of AbstractArray, it can be used with types that override broadcast behaviour such as CuArray.

For DistributedArrays I solved this by using nested BroadcastStyles.

**Comment by mbauman** [comment]
Yeah, I imagine this working the way the current ad-hoc [`broadcast` methods](https://github.com/JuliaLang/julia/blob/c630477c4e6af0ae04163378571587f82f6178c8/stdlib/LinearAlgebra/src/adjtrans.jl#L230-L231) do — just unwrap all the arguments, re-call broadcast on the unwrapped arrays (which will compute the "nested" style) and then appropriately re-wrap.
